### PR TITLE
Enhancement Run Instances

### PR DESCRIPTION
**Pull request by Tirla-Alin**
The input mapping logic was not according to the Amazon API; It was required (from our code) to specify at least one private ip address for the network interface in order to be able to request a public ip address.

o	Signed-off by Tirla Alin alin.tirla@hpe.com

### CODE REVIEW

**Comment by mihaitusa** [comment]
@Tirla-Alin Are you sure about it? Because (as far as I remember) the private IP address was set only if privateIpAdressesString is not blank...

**Comment by Tirla-Alin** [comment]
@mihaitusa all fields are optional. But if provided they should have the same length (if input is empty we simply ignore it).

**Comment by mihaitusa** [comment]
@Tirla-Alin Okay... just make sure that the scenarios proposed by AWS are working...

**Comment by Bonczidai** [comment]
@Tirla-Alin from my POV looks ok

**Comment by Tirla-Alin** [comment]
I will add Unit Tests as been required for these new functions i provided.

**Comment by Tirla-Alin** [comment]
Added unit tests. You can resume the review.

File: cs-amazon/src/test/java/io/cloudslang/content/amazon/execute/QueryApiExecutorTest.java [hunk 1]
```diff
@@ -353,6 +370,84 @@ public void testRebootInstances() throws Exception {
     }
 
     @Test
+    public void testRunInstances() throws Exception {
+        toTest.execute(getCommonInputs("RunInstances", HEADERS, ""), getRunInstancesCustomInputs(), getRunInstancesEbsInputs(), getRunInstancesElasticIpInputs(), getRunInstancesIamInputs(), getRunInstancesInstanceInputs(), getRunInstancesNetworkInputs());
```
Comment (Bonczidai) [inline cs-amazon/src/test/java/io/cloudslang/content/amazon/execute/QueryApiExecutorTest.java]: I suggest to break such long lines

File: cs-amazon/src/test/java/io/cloudslang/content/amazon/execute/QueryApiExecutorTest.java [hunk 2]
```diff
@@ -371,6 +388,84 @@ public void testRebootInstances() throws Exception {
...
+
+        verify(amazonSignatureServiceMock, times(1)).signRequestHeaders(any(InputsWrapper.class), eq(getHeadersMap()),
```
Comment (Bonczidai) [inline cs-amazon/src/test/java/io/cloudslang/content/amazon/execute/QueryApiExecutorTest.java]: just make sure `any` matcher is appropriate here

Comment (Tirla-Alin) [inline cs-amazon/src/test/java/io/cloudslang/content/amazon/execute/QueryApiExecutorTest.java]: We don't pass the (InputsWrapper) as parameter to this function, and the values from it are tested that are passed correctly after this with eq(getQueryParamsMap("RunInstances")) .

**Comment by tethryus** [comment]
👍 @Tirla-Alin Since the reviews have passed, this can be merged and released to MC.

**Pull request merged**